tools/online_editor/src/editor_widget.ts

```diff
@@ -440,6 +474,19 @@ class EditorPaneWidget extends Widget {
     const model = handle as monaco.editor.ITextModel;
     return model.getValue().substring(start, end);
   }
+
+  async read_from_url(url: string): Promise<string> {
+    let response = await fetch(url);
+    let text = await response.text();
```

Comment:
It seems wrong to me to unconditionally run `fetch()`. Shouldn't this only be done between line 482 and 483?

(Unless you expect this function to always fetch but then it should update the model in line 487?)

Reply:
True, I should check before the fetch, too.

The second check is still necessary as the LSP and preview both request the download and like to race each other.